### DISCUSSION TITLE
Incorrect Mastodon account id on profile page

### DISCUSSION THREAD

**Discussion by nielsbasjes** [community/community#55036] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Bug

### Body

On my profile page I can now add a link to my mastodon account as I have done: https://github.com/nielsbasjes .
My mastodon account name is `@niels@basjes.nl` which is what people can use and see when they look me up on Mastodon.

I am hosting the actual mastodon server on `https://mastodon.basjes.nl/` (so NOT on `https://basjes.nl`) which is a completely valid and supported setup for mastodon servers. 
See https://docs.joinmastodon.org/admin/config/#basic about the config settings LOCAL_DOMAIN and WEB_DOMAIN.

So given the options available on Github I had to configure `https://mastodon.basjes.nl/@niels` which is ok.

**BUG:** Github shows the mastodon account name to visitors of that page as `@niels@mastodon.basjes.nl` instead of `@niels@basjes.nl`.

**Improvement** I would like to simple enter my real account name `@niels@basjes.nl` and from there Github would be able to retrieve the actual mastodon instance and account homepage `https://mastodon.basjes.nl/@niels`

_Some background on how Mastodon does this:_

An actual mastodon client/server finds the server and homepage of a user by means of webfinger  (see https://docs.joinmastodon.org/spec/webfinger/ ). So a mastodon client/server goes from `@niels@basjes.nl` to getting this URL `https://basjes.nl/.well-known/webfinger?resource=acct:niels@basjes.nl` which returns the appropriate account information (i.e. homepage url and such).

Seems link Github is not using the webfinger to get the account name and is making assumptions instead.
If you request

    curl https://mastodon.basjes.nl/.well-known/webfinger?resource=acct:niels@mastodon.basjes.nl

or 

    curl -L https://basjes.nl/.well-known/webfinger?resource=acct:niels@basjes.nl

you get the correct information (snippet):

    {
       "subject":"acct:niels@basjes.nl",
       "aliases":[
            "https://mastodon.basjes.nl/@niels",
            "https://mastodon.basjes.nl/users/niels"
       ], ...

which clearly shows the real account id using the mastodon way of doing this.

**Comment by 100terres** [comment]
I'm facing the exact same issue on my GitHub profile.
